crates/searchers/openapi.yaml

```diff
@@ -0,0 +1,584 @@
+        A UniswapV3-like 2-token concentrated liquidity pool.
+      type: object
+      required:
+        - kind
+        - tokens
+        - sqrtPrice
+        - liquidity
+        - tick
+        - liquidityNet
+        - fee
+      properties:
+        kind:
+          type: string
+          enum: [concentratedliquidity]
+        tokens:
+          type: array
+          items:
+            $ref: "#/components/schemas/Address"
+        sqrtPrice:
+          $ref: "#/components/schemas/BigInt"
+        liquidity:
+          $ref: "#/components/schemas/TokenAmount"
+        tick:
+          $ref: "#/components/schemas/BigInt"
+        liquidityNet:
+          type: object
+          additionalProperties:
+            $ref: "#/components/schemas/TokenAmount"
+          description: |
+            A map of tick indices to there liquidity values.
+        fee:
+          $ref: "#/components/schemas/Decimal"
+
+    ForeignLimitOrder:
+      description: |
+        A 0x-like liquidity order external to CoW Protocol.
+      type: object
+      required:
+        - kind
+        - makerToken
+        - takerToken
+        - makerAmount
+        - takerAmount
+        - takerTokenFeeAmount
+      properties:
+        kind:
+          type: string
+          enum: [limitorder]
+        hash:
+          $ref: "#/components/schemas/Digest"
+        makerToken:
+          $ref: "#/components/schemas/Address"
+        takerToken:
+          $ref: "#/components/schemas/Address"
+        makerAmount:
+          $ref: "#/components/schemas/TokenAmount"
+        takerAmount:
+          $ref: "#/components/schemas/TokenAmount"
+        takerTokenFeeAmount:
+          $ref: "#/components/schemas/TokenAmount"
+
+    LiquidityParameters:
+      oneOf:
+        - $ref: "#/components/schemas/ConstantProductPool"
+        - $ref: "#/components/schemas/WeightedProductPool"
+        - $ref: "#/components/schemas/StablePool"
+        - $ref: "#/components/schemas/ConcentratedLiquidityPool"
+        - $ref: "#/components/schemas/ForeignLimitOrder"
+
+    Liquidity:
+      description: |
+        On-chain liquidity that can be used in a solution. This liquidity is
+        provided to facilitate TODO
+      allOf:
+        - $ref: "#/components/schemas/LiquidityParameters"
+        - type: object
+          properties:
+            address:
+              $ref: "#/components/schemas/Address"
+            gasEstimate:
+              type: integer
+              description: |
+                A rough approximation in the number gas units required to
+                use this liquidity on-chain.
+
+    Auction:
+      description: |
+        The abstract auction instance in order to solve.
+      type: object
+      properties:
+        id:
+          type: integer
+          description: |
+            The identifier for the auction. Will be set to `null` for requests
+            that are not part of an auction (when quoting token prices for
+            example).
```

Comment:
Turned this into a `string` so that it can be an opaque ID. 

I also created https://github.com/cowprotocol/services/issues/973 so we get back to this eventually, but I don't want to derail the conversation about the API too much on this small-ish implementation detail.

Of course, if you disagree and think this is important for the API - feel free to unresolve the issue.